### PR TITLE
[losses] migrate npairs multilabel loss

### LINKED ISSUE #26: Implement Metric Losses
Per the [RFC](https://github.com/tensorflow/community/blob/393dfc9e03dd4150a839de6357a26e0259aa97d6/rfcs/20181214-move-to-addons.md), we need to move [metric_losses](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/contrib/losses/python/metric_learning) from contrib to addons:

- [x] pairwise_distance
- [x] contrastive_loss
- [x] triplet_semihard_loss
- [x] npairs_loss
- [ ] npairs_loss_multilabel
- [x] lifted_struct_loss

This will involve inheriting from base Keras Loss, modifying the code to match those APIs, and modifying test cases to run in TF2.

### PR DESCRIPTION
**Pull request by WindQAQ**
Closes #26. Finally the last one :-)

- Migrate `npairs_loss_multilabel` from contrib
- Rename `npairs_loss_multilabel` to `npairs_multilabel_loss`, which makes sense to addons' naming style
- Modify input spec from
    - `labels` (list of sparse tensor)
    - `embedding_anchor`
    - `embedding_posititve`

    to
    - `y_true` (either tensor or sparse tensor)
    - `y_pred` (similarity matrix, which is the multiplication of anchor and positive)

### CODE REVIEW

**Comment by facaiy** [comment]
@Squadrick Hi, Dheeraj. I think the PR looks good. Do you have time to double check it? Thanks :-)

**Comment by WindQAQ** [comment]
> Looks good, Tzu-wei!
> 
> By the way, can we provide a helper method which calculates `similarity_matrix`? It might be user-friendly for tf 1.x users. What do you think?

Seems good, but it will be a one line function like `tf.matmul(a, b)`. Previously, we inherit from `LossFunctionWrapper`, and that's why we have input signature `y_true` and `y_pred` only, resulting in computation of similarity matrix outside the loss function. But now, we inherit from `Loss`, I wonder if we could use signature `label`, `embedding_anchor` and `embedding_positive`, which makes much sense to me in the field of metric learning. What do you feel about this?

**Comment by facaiy** [comment]
@WindQAQ Oh, you're right, forget it

**Comment by Squadrick** [comment]
Maybe add a test to check that the new loss is compatible with the Keras Model API? It would be great if we add it under `utils` and every loss has to use the check. What do y'all think?

**Comment by facaiy** [comment]
Like `layer_test` or https://github.com/tensorflow/tensorflow/blob/0197a2d8a3070af763cb67227835ee63df095e6d/tensorflow/python/keras/losses_test.py#L68 ? I have no objection. How about filing an issue and finishing it in next PR?
